test/integration/shoots/applications/shoot_app_test.go

```diff
@@ -0,0 +1,287 @@
+		Expect(err).To(BeNil())
+
+		err = shootTestOperations.WaitUntilPodIsRunning("redis-master-0", helmDeployNamespace, "shoot", 10*time.Second, 3600*time.Second)
+		Expect(err).To(BeNil())
+
+		redisSlaveLabels := map[string]string{
+			"app":  "redis",
+			"role": "slave",
+		}
+		err = shootTestOperations.WaitUntilPodWithLabelsIsReady(redisSlaveLabels, helmDeployNamespace, "shoot", 10*time.Second, 3600*time.Second)
+		Expect(err).To(BeNil())
+
+		guestBookParams := struct {
+			HelmDeployNamespace string
+			ShootDNSHost        string
+		}{
+			helmDeployNamespace,
+			fmt.Sprintf("guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain),
+		}
+
+		// Parse the guestbook app template
+		var writer bytes.Buffer
+		err = guestBooktpl.Execute(&writer, guestBookParams)
+		Expect(err).To(BeNil())
+
+		// Apply the guestbook app resources to shoot
+
+		manifestReader := kubernetes.NewManifestReader(writer.Bytes())
+		err = shootTestOperations.K8sShootClient.Applier().ApplyManifest(context.TODO(), &manifestReader)
+		Expect(err).To(BeNil())
+
+		// define guestbook app urls
+		guestBookAppURL := fmt.Sprintf("http://guestbook.ingress.%s", *shootTestOperations.Shoot.Info.Spec.DNS.Domain)
+		pushUrl := fmt.Sprintf("%s/rpush/guestbook/foobar-%s", guestBookAppURL, shootTestOperations.Shoot.Info.Name)
+		pullUrl := fmt.Sprintf("%s/lrange/guestbook", guestBookAppURL)
+
+		// Check availability fo the guestbookapp
+		err = shootTestOperations.WaitUntilGuestbookAppIsAvailable([]string{guestBookAppURL, pushUrl, pullUrl}, 10*time.Second, 1200*time.Second)
+		Expect(err).To(BeNil())
+
+		// Push foobar-<shoot-name> to the guestbook app
+		_, err = shootTestOperations.HTTPGet(pushUrl)
+		Expect(err).To(BeNil())
+
+		// Pull foobar
+		pullResponse, err := shootTestOperations.HTTPGet(pullUrl)
+		Expect(err).To(BeNil())
+
+		responseBytes, err := ioutil.ReadAll(pullResponse.Body)
+		Expect(err).To(BeNil())
+
+		// test if foobar-<shoot-name> was pulled successfully
+		bodyString := string(responseBytes)
+		Expect(bodyString).To(ContainSubstring(fmt.Sprintf("foobar-%s", shootTestOperations.Shoot.Info.Name)))
+
+		shootTestOperations.ShootGardenerTest.Logger.Info("Guestbook app was deployed successfully!")
+
+	})
+	FContext("Network Policy Testing", func() {
+
+		var (
+			dashboardIP, nodeIP string
+			err                 error
+		)
+		It("Should install network connectivity tools", func() {
+			// first we need to install the testing tools
+			err := shootTestOperations.PodExecByLabel(apiserverLabels, "kube-apiserver",
+				"apt-get update && apt-get -y install curl netcat", "seed")
+			Expect(err).To(BeNil())
+		})
+
+		// we start our network connections tests
+		DescribeTable("#APIServerConnectivity",
+			func(host, port string, expected int) {
+				// FIXME Later: Ugly hack, currently Gingko does not support variable setting for Entries (https://github.com/onsi/ginkgo/issues/378)
+				switch port {
+				case "8443":
```

Comment:
Don't do switches like that, this is an indicator that you should create different tests.